FurkannOzbek/application-hub
# [FEATURE] Implement Text Editor for Application Notes

# [FEATURE] Implement Text Editor for Application Notes

---

## Description

This PR implements a rich text editor component for managing application notes using React-Quill. The component provides a user-friendly interface for creating, editing, and viewing notes associated with job applications.
Also, the EmptyState component was added to be used with all tabs which don't have content.

Problem: Need a way to maintain detailed notes for each job application with rich text formatting
Impact: Users can now create and maintain formatted notes with features like headers, lists, and text styling

---

## Changes Made

### Frontend:
Notes Component:
- Implemented rich text editor using React-Quill
- Added edit/read modes with proper state management
- Implemented empty state display

Editor Styling:
- Added comprehensive theme-aware styling for the editor
- Implemented hover states and active states for toolbar buttons
- Ensured consistent styling with the application's theme

Empty State Component:
- Created a separate component for empty state display
- Consistent styling with other empty states in the application

---

## Screenshots (if applicable)

### Empty state
![screencapture-localhost-3000-applications-28-2024-11-24-09_54_54](https://github.com/user-attachments/assets/72e160fa-e116-4cf9-9b4a-eec017edc15b)

![screencapture-localhost-3000-applications-28-2024-11-24-09_54_54](https://github.com/user-attachments/assets/c5a8ff63-19d6-4d49-aad1-1f040330e784)


### Editor in read mode
![screencapture-localhost-3000-applications-39-2024-11-21-20_53_08](https://github.com/user-attachments/assets/858f608b-553a-4779-8da1-a191357a6c4f)

![screencapture-localhost-3000-applications-39-2024-11-21-20_52_58](https://github.com/user-attachments/assets/e70d21fd-b73e-4d46-b729-326b582f977c)

### Editor in edit mode with toolbar
![screencapture-localhost-3000-applications-39-2024-11-21-20_52_30](https://github.com/user-attachments/assets/712fe587-7342-40dd-ac66-80d61310809c)

![screencapture-localhost-3000-applications-39-2024-11-21-20_52_47](https://github.com/user-attachments/assets/18ed4513-fd16-4488-845d-6d4818214716)

### On mobile:
<img src="https://github.com/user-attachments/assets/608f7a47-c71a-4c8f-aeb6-69b29eedda43" alt="Mobile" width="200" />
<img src="https://github.com/user-attachments/assets/f27bb452-afa3-407c-954d-cc5354e31215" alt="Mobile" width="200" />
<img src="https://github.com/user-attachments/assets/315bec19-a6c0-41fe-8550-86ffe422c496" alt="Mobile" width="200" />


---

## Testing Steps

1. Navigate to any application's details page
2. Click on the Notes tab
3. Verify empty state is displayed correctly
4. Test editor functionality:
 - Click Add button
 - Try different text formatting options
 - Save changes
 - Cancel changes
5. Test responsive behavior on different screen sizes
6. Verify theme consistency in both light and dark modes
---

## Issues Addressed

Closes [Trello ticket link for Notes feature](https://trello.com/c/CNimWft4)

---

## Additional Notes

The next step is creating endpoints to get, put, and update notes from the Database.
